### PR TITLE
[GC-25] Implent CrewAI + RESTAPI with Flask

### PR DESCRIPTION
**Pull request by nutkung1**
## Summary by CodeRabbit

- **New Features**
  - Introduced functionality for managing persistent storage in the vector store.
  - Implemented a document processing and retrieval system for improved text ingestion and embedding.
  - Developed an AI-driven multi-agent architecture for handling user queries, enhancing conversational capabilities.
  - Established a robust question-answering framework with automated content generation and validation.

- **Bug Fixes**
  - Addressed issues relating to data storage and retrieval processes.

- **Documentation**
  - Enhanced comments and code documentation to improve understanding of new features and functionalities.

### CODE REVIEW

File: deletepersist.py
Comment (Gunn-Wangwichit) [inline deletepersist.py:1]: You do not need to use this file, remove it

File: embedded.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+vectordb = Chroma.from_documents(documents=texts, 
+                                 embedding=embeddings,
+                                 persist_directory=persist_directory)
+
+vectordb.persist()
+vectordb = None
+vectordb = Chroma(persist_directory=persist_directory, 
+                  embedding_function=embeddings)
+
+# retriever = vectordb.as_retriever()
```
Comment (Gunn-Wangwichit) [inline embedded.py:25]: remove this line, you did not use it

File: embedded.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+print("Successkung")
```
Comment (Gunn-Wangwichit) [inline embedded.py]: Wtf is Successkung
```suggestion
print("Success")
```

File: test.py
Comment (Gunn-Wangwichit) [inline test.py:1]: Merge the Flask part to the model.py

Comment (Gunn-Wangwichit) [inline test.py:1]: Pls remove unnecessary  comments part

**Review by Gunn-Wangwichit — COMMENTED** [review]
Pls push only the main part (embbedded and model). Other parts that remove them

File: model.py [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+)
+
+
+# Define the Crew with memory enabled
+rag_crew = Crew(
+    agents=[Retriever_Agent, content_writer_agent, hallucination_checker],
+    tasks=[retriever_task, content_writer_task, hallucination_task],
+    verbose=True,  # Enable memory for the entire crew
+    process=Process.sequential,
+    # memory=True,
```
Comment (Billy19191) [inline model.py]: If you don't use this, please remove but in case for further config, it is fine to leave it as it

**Review by Gunn-Wangwichit — APPROVED** [review]
Good to go

**Pull request merged**